ethereum-consensus/src/deneb/polynomial_commitments.rs

```diff
@@ -0,0 +1,138 @@
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct KzgProof(ByteVector<BYTES_PER_PROOF>);
+
+fn blob_to_kzg_commitment(blob: Blob, kzg_settings: &KzgSettings) -> Result<KzgCommitment, Error> {
+    let inner = &blob.0;
+    let blob = c_kzg::Blob::from_bytes(inner.as_ref()).unwrap();
+
+    let commitment = c_kzg::KzgCommitment::blob_to_kzg_commitment(&blob, kzg_settings)?;
+    let inner = ByteVector::try_from(commitment.to_bytes().as_slice()).unwrap();
+
+    Ok(KzgCommitment(inner))
+}
+
+fn compute_kzg_proof(
+    blob: Blob,
+    z_bytes: Bytes32,
+    kzg_settings: &KzgSettings,
+) -> Result<(KzgProof, Bytes32), Error> {
+    let inner = blob.0.as_ref();
+    let blob = c_kzg::Blob::from_bytes(inner).unwrap();
+
+    let (proof, evaluation) = c_kzg::KzgProof::compute_kzg_proof(&blob, &z_bytes, kzg_settings)?;
+    let proof = ByteVector::try_from(proof.to_bytes().as_ref()).unwrap();
+
+    Ok((KzgProof(proof), evaluation))
+}
+
+/// This function doesn't verify that the commitment is correct with respect to the blob.
+fn compute_blob_kzg_proof(
+    blob: Blob,
+    commitment_bytes: Bytes48,
+    kzg_settings: &KzgSettings,
+) -> Result<KzgProof, Error> {
+    let inner = blob.0.as_ref();
+    let blob = c_kzg::Blob::from_bytes(inner).unwrap();
+
+    let ckzg_proof =
+        c_kzg::KzgProof::compute_blob_kzg_proof(&blob, &commitment_bytes, kzg_settings)?;
+
+    let bytes_proof = ckzg_proof.to_bytes();
+    let proof = ByteVector::try_from(bytes_proof.as_ref()).unwrap();
+
+    Ok(KzgProof(proof))
+}
+
+fn verify_kzg_proof(
+    commitment_bytes: Bytes48,
+    z_bytes: Bytes32,
+    y_bytes: Bytes32,
+    proof_bytes: Bytes48,
+    kzg_settings: &KzgSettings,
+) -> Result<bool, Error> {
+    let out = c_kzg::KzgProof::verify_kzg_proof(
+        &commitment_bytes,
+        &z_bytes,
+        &y_bytes,
+        &proof_bytes,
+        &kzg_settings,
+    )?;
+
+    Ok(out)
+}
+
+fn verify_blob_kzg_proof(
+    blob: Blob,
+    commitment_bytes: Bytes48,
+    proof_bytes: Bytes48,
+    kzg_settings: &KzgSettings,
+) -> Result<bool, Error> {
+    let bytes = blob.0.as_ref();
+    let blob = c_kzg::Blob::from_bytes(bytes).unwrap();
+
+    let out = c_kzg::KzgProof::verify_blob_kzg_proof(
+        &blob,
+        &commitment_bytes,
+        &proof_bytes,
+        &kzg_settings,
+    )?;
+
+    Ok(out)
+}
+
+fn verify_blob_kzg_proof_batch(
+    blobs: &[Blob],
+    commitments_bytes: &[Bytes48],
+    proofs_bytes: &[Bytes48],
+    kzg_settings: &KzgSettings,
+) -> Result<bool, Error> {
+    let mut c_kzg_blobs = Vec::with_capacity(blobs.len());
+
+    for bytes in blobs.iter().map(|blob| blob.0.as_ref()) {
+        let blob = c_kzg::Blob::from_bytes(bytes)?;
+        c_kzg_blobs.push(blob);
+    }
+
+    let out = c_kzg::KzgProof::verify_blob_kzg_proof_batch(
+        &c_kzg_blobs,
+        commitments_bytes,
+        proofs_bytes,
+        kzg_settings,
+    )?;
+
+    Ok(out)
```

Comment:
more of an FYI but you can also do stuff like this
```suggestion
    c_kzg::KzgProof::verify_blob_kzg_proof_batch(
        &c_kzg_blobs,
        commitments_bytes,
        proofs_bytes,
        kzg_settings,
    ).map_err(Into::into)
```